app/utils/ReceiptDataHelper.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import connectors.VatSubscriptionConnector
+import javax.inject.{Inject, Singleton}
+import models.SubmitVatReturnModel
+import models.auth.User
+import models.errors.{BadRequestError, HttpError}
+import models.nrs.{Declaration, _}
+import play.api.{Logger, Play}
+import play.api.i18n.MessagesApi
+import play.api.mvc.AnyContent
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+@Singleton
+class ReceiptDataHelper @Inject()(
+                                   vatSubscriptionConnector: VatSubscriptionConnector,
+                                   implicit val messages: MessagesApi
+                                 ) {
+
+  def extractReceiptData(submitModel: SubmitVatReturnModel)
+                        (implicit user: User[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Either[HttpError, ReceiptData]] = {
+
+    val language: Language = user.cookies.get(Play.langCookieName) match {
+      case Some(cookieValue) => Language.fromString(cookieValue.value)
+      case None => EN
+    }
+
+    extractDeclaration(submitModel).map {
+      case Right(declaration) =>
+        Right(ReceiptData(
+          language,
+          extractAnswers(submitModel),
+          declaration
+        ))
+      case Left(error) => Left(error)
+    }
+  }
+
+  private def extractAnswers(submitModel: SubmitVatReturnModel)(implicit user: User[_]): Seq[Answers] = {
+    val boxSixSearchKey = if (submitModel.flatRateScheme) "boxSixFlatRate" else "boxSixNoFlatRate"
+
+    val answerSeq = Seq(
+      ("box1", messages("confirm_submission.boxOneDescription"), submitModel.box1),
+      ("box2", messages("confirm_submission.boxTwoDescription"), submitModel.box2),
```

Comment:
I'm not sure the language preference is being passed into the message key lookup and pulling out welsh messages when the cookie is set - otherwise wouldn't the tests fail when comparing the english in `expectedAnswers()` in the welsh scenario?